src/blob.rs

```diff
@@ -1,86 +1,258 @@
-use std::path::Path;
+use futures::{
+    future::Future,
+    task::{Poll, Waker},
+};
+use std::{
+    collections::BTreeMap,
+    fs, io,
+    os::unix::fs::FileExt,
+    path::{Path, PathBuf},
+    pin::Pin,
+};
 
-use crate::record::Record;
+use crate::record::{self, Record};
+
+const BLOB_MAGIC_BYTE: u64 = 0xdeaf_abcd;
+
+type Result<T> = std::result::Result<T, Error>;
 
 /// A `Blob` struct for performing of database,
-#[derive(Debug, Default)]
-pub struct Blob<T> {
+#[derive(Debug)]
+pub struct Blob {
+    index: BTreeMap<Vec<u8>, record::Header>,
```

Comment:
One more question: what ordering function is used for `Vec<u8>`? I don't see any implementation here.

Reply:
[```Vec::ord(_)```](https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Ord)